src/app/query/global-query/search-and-schemas-section/search-and-schemas-section.component.ts

```diff
@@ -0,0 +1,145 @@
+import { parseCurrentSchema } from "src/app/common/app.helpers";
+import { NgbTypeaheadSelectItemEvent } from "@ng-bootstrap/ng-bootstrap";
+import { MaybeNull } from "src/app/common/app.types";
+import { DatasetSchema } from "src/app/interface/dataset.interface";
+import { DatasetAutocompleteItem, TypeNames } from "src/app/interface/search.interface";
+import { SearchApi } from "src/app/api/search.api";
+import { DatasetService } from "src/app/dataset-view/dataset.service";
+import AppValues from "src/app/common/app.values";
+import { SqlQueryService } from "src/app/services/sql-query.service";
+
+@Component({
+    selector: "app-search-and-schemas-section",
+    templateUrl: "./search-and-schemas-section.component.html",
+    styleUrls: ["./search-and-schemas-section.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchAndSchemasSectionComponent extends BaseComponent implements OnInit {
+    private searchResult: GlobalQuerySearchItem[] = [];
+    public inputDatasets = new Set<string>();
+    public searchDataset = "";
+    private readonly delayTime: number = AppValues.SHORT_DELAY_MS;
+
+    private cdr = inject(ChangeDetectorRef);
+    private datasetService = inject(DatasetService);
+    private appSearchAPI = inject(SearchApi);
+    private sqlQueryService = inject(SqlQueryService);
+
+    public ngOnInit(): void {
+        this.sqlQueryService.sqlQueryResponseChanges
+            .pipe(
+                switchMap((sqlQueryResponse: MaybeNull<SqlQueryResponseState>) => {
+                    if (sqlQueryResponse) {
+                        const schemaResponses = sqlQueryResponse.involvedDatasetsId.map((id: string) =>
+                            this.datasetService.requestDatasetSchema(id),
+                        );
+                        return this.processSchemaResponses(schemaResponses);
+                    }
+                    return of(null);
+                }),
+                takeUntilDestroyed(this.destroyRef),
+            )
+            .subscribe((data: MaybeNull<GlobalQuerySearchItem[]>) => {
+                if (data) {
+                    this.searchResult = data;
+                    this.cdr.detectChanges();
+                }
+            });
+    }
+
+    private processSchemaResponses(
+        schemaResponses: Observable<GetDatasetSchemaQuery>[],
+    ): Observable<GlobalQuerySearchItem[]> {
+        return combineLatest(schemaResponses).pipe(
+            map((datasets: GetDatasetSchemaQuery[]) => {
+                const result = datasets.map((item: GetDatasetSchemaQuery) => {
+                    const datasetAlias = item.datasets.byId?.alias;
+                    const schema = parseCurrentSchema(item.datasets.byId?.metadata.currentSchema);
+                    return { datasetAlias, schema } as GlobalQuerySearchItem;
+                });
+                return result;
+            }),
+        );
+    }
+
+    public removeDataset(datasetAlias: string): void {
```

Comment:
You can pass the item directly from HTML template instead of searching for it

Reply:
There is no big difference, you still need to search by alias and delete it